linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/factory/SparkEngineConnResourceFactory.scala

```diff
@@ -20,19 +20,50 @@ package org.apache.linkis.engineplugin.spark.factory
 import org.apache.linkis.common.utils.{ByteTimeUtils, Logging}
 import org.apache.linkis.engineplugin.spark.config.SparkResourceConfiguration._
 import org.apache.linkis.manager.common.entity.resource.{
+  DriverAndKubernetesResource,
   DriverAndYarnResource,
+  KubernetesResource,
   LoadInstanceResource,
   Resource,
   YarnResource
 }
-import org.apache.linkis.manager.engineplugin.common.resource.AbstractEngineResourceFactory
+import org.apache.linkis.manager.engineplugin.common.resource.{
+  AbstractEngineResourceFactory,
+  EngineResourceRequest
+}
+import org.apache.linkis.manager.label.entity.cluster.ClusterLabel
+import org.apache.linkis.manager.label.utils.LabelUtil
 
 import org.apache.commons.lang3.StringUtils
 
 import java.util
 
+import io.fabric8.kubernetes.api.model.Quantity
+
 class SparkEngineConnResourceFactory extends AbstractEngineResourceFactory with Logging {
 
+  override protected def getMinRequestResource(
+      engineResourceRequest: EngineResourceRequest
+  ): Resource = {
+    val clusterLabel = LabelUtil.getLabelFromList[ClusterLabel](engineResourceRequest.labels)
+    if (clusterLabel != null && clusterLabel.getClusterType == "K8S") {
```

Comment:
change '==' to `equals`， and use `toUpperCase` to be compatible with string matching